src/statement.d

```diff
@@ -1893,6 +1894,45 @@ extern (C++) final class SwitchStatement : Statement
         return true;
     }
 
+    final bool checkLabel()
+    {
+        bool checkVar(VarDeclaration vd)
+        {
+            if (!vd || vd.isDataseg() || (vd.storage_class & STCmanifest))
+                return false;
+
+            VarDeclaration last = lastVar;
+            while (last && last != vd)
+                last = last.lastVar;
+            if (last == vd)
+            {
+                // All good, the label's scope has no variables
+            }
+            else if (vd.ident == Id.withSym)
+            {
+                error("switch skips declaration of with temporary at %s", vd.loc.toChars());
```

Comment:
This string is confusing because "with" is not clearly marked as a keyword. How about `"switch skips declaration of temporary in \"with\" statement at %s"``